### DISCUSSION TITLE
while deploying a simple Solidity contract to zkSync Testnet using the zkSync Remix plugin.Getting an error

### DISCUSSION THREAD

**Discussion by Phinihas** [Cyfrin/foundry-full-course-cu#2851] [Q&A]
The error is below 
Deploying contract SimpleStorage
Error: "underlying network changed (event=\"changed\", network={\"chainId\":300,\"name\":\"zkSync Sepolia Testnet\"}, detectedNetwork={\"name\":\"sepolia\",\"chainId\":11155111,\"ensAddress\":null}, code=NETWORK_ERROR, version=providers/5.7.2)"

The code is below
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;
// use versions between 0.8.19 and 0.9.0 (excluded) 
// pragma solidity ^0.8.19; 
// pragma solidity >=0.8.19 <0.9.0;

contract SimpleStorage {
    // Basic types 
    // bool hasFavoriteNumber = true; 
    // uint256 favoriteNumber = 88; 
    // string favoriteNumberInText="eighty-eight"; 
    // int256 favoriteInt=-88; 
    // address myAddress=0xb3808A4F4a85f88D9f135a563e15a10E6F4796e4; 
    // bytes32 favoriteBytes32="cat";

    uint256 public myfavoriteNumber;
    
    struct Person {
    uint256 my_favorite_number;
    string name;
}
    Person[] public listOfPeople;

    mapping (string => uint256) public nameToFavoriteNumber;
    mapping (uint256 => string) public favoriteNumberToName;

   function store(uint256 _favoriteNumber) public {
    myfavoriteNumber = _favoriteNumber;
    //uint256 testVar = 5;
}

function something() public {
   //testVar = 6; // will raise a compilation error
   myfavoriteNumber = 7; // this can be accessed because it's in the main contract scope
}

    function add_person(string memory _name, uint256 _favorite_number) public {
        listOfPeople.push(Person(_favorite_number, _name));
        nameToFavoriteNumber[_name]=_favorite_number;
        favoriteNumberToName[_favorite_number]=_name;

}

}

**Comment by heheboii11** [comment]
Can you share your ```deployscript``` and the terminal command you are using to deploy the script( starts with ```forge script```) and also check your ```.env``` , whether it is provided with the necessary ```RPC-URL```,  ```private-key``` and ```etherscan-api```(if needed). 
Please do not share your ```.env```

**Comment by EngrPips** [comment]
Hello @Phinihas, You need to switch your `Metamask` network to `ZKSync-sepolia` for the deployment process to go through successfully. You can look at the screenshot below to get a hint on how to switch your `Metamask` network 

<img width="357" alt="Screenshot 2024-11-12 at 08 01 19" src="https://github.com/user-attachments/assets/8c587d13-8c84-496f-b432-28733e15aa0c">

<img width="357" alt="Screenshot 2024-11-12 at 08 02 49" src="https://github.com/user-attachments/assets/c2d7f19c-78b0-4b88-bf7a-4dc3a8fa46b6">

**Reply by heheboii11** [reply]
This is the solution, I did not see you are coding in ```remix```. My bad, sorry for wrong reply 😅

**Reply by Phinihas** [reply]
After switch your Metamask network to ZKSync-sepolia still getting same error.
![Screenshot 2024-11-12 143524](https://github.com/user-attachments/assets/693ba621-a071-4e52-829a-5ccab5db7742)
![Screenshot 2024-11-12 143542](https://github.com/user-attachments/assets/d3dd0f1c-144f-4314-bea8-b57b72148c3a)

**Reply by EngrPips** [reply]
Maybe clear your browser data and retry. Also, do you put your files inside a `contracts` folder?

**Reply by Phinihas** [reply]
I added SimpleStorage contract file inside contracts folder but still getting same error as below

Error: "underlying network changed (event=\"changed\", network={\"chainId\":300,\"name\":\"zkSync Sepolia Testnet\"}, detectedNetwork={\"name\":\"sepolia\",\"chainId\":11155111,\"ensAddress\":null}, code=NETWORK_ERROR, version=providers/5.7.2)"

**Reply by EngrPips** [reply]
It seems like the network is not changing correctly. can you close the page and restart the process.
